### ISSUE TITLE
`fltk-config` returns Linux LDFLAGS

### ISSUE THREAD

**Issue by mmuetzel** [msys2/MINGW-packages#8806]
The `fltk-config` program that is installed as part of `mingw-w64-x86_64-fltk` returns the following:
```
$ fltk-config --use-gl --ldflags
-L/mingw64/lib -lfltk_gl -lGLU -lGL -lfltk -lpthread
```
AFAICT, there are no GLU or GL libraries on Windows. Instead of `-lGLU -lGL` these flags should probably contain `-lopengl32 -lglu32`.

That causes linker issues when trying to compile projects that use `fltk-config` to get the linker flags for FLTK with OpenGL support.

It would be nice if `fltk-config` in MSYS2 could be patched to return flags that can be used for successful linking on MinGW.
Or is this something that should be fixed in upstream FLTK?

I haven't tested that. But the same issue probably also occurs with the `fltk-config` that is installed with `mingw-w64-i686-fltk`.

**Comment by Biswa96** [comment]
Seems like an upstream issue.

**Comment by lazka** [comment]
Usually upstream doesn't support the config shell scripts on Windows.. since that only works in MSYS2 and nowhere else.

**Comment by lazka** [comment]
One option is also to remove it, assuming there are fallbacks like pkg-config and users of that library support that.

**Comment by mmuetzel** [comment]
I can't find a pkg-config file for fltk.

Another option might be to install symbolic links for those libraries at the location `fltk-config` claims them to be:
```
ln -sf /mingw64/x86_64-w64-mingw32/lib/libglu32.a /mingw64/lib/libGLU.a
ln -sf /mingw64/x86_64-w64-mingw32/lib/libopengl32.a /mingw64/lib/libGL.a
```
I haven't tested that. And I can't tell if this could mess up something else though.

**Comment by revelator** [comment]
this is caused by us using the cmake build feature instead of autotools which have these options correct, heres a patch that adds the correct opengl libraries to the cmake build.

[0004-correct-opengl-libraries-win32.zip](https://github.com/msys2/MINGW-packages/files/6643113/0004-correct-opengl-libraries-win32.zip)

**Comment by mmuetzel** [comment]
Thank you.
Is there a way I can test your patch?

**Comment by revelator** [comment]
add it to the PKGBUILD file along the other patches, then run updpkgsums before building.

**Comment by mmuetzel** [comment]
Thank you.
I made this change to PKGBUILD (with the help of `updpkgsums` like you suggested):
```
diff --git "a/mingw-w64-fltk/PKGBUILD" "b/mingw-w64-fltk/PKGBUILD"
index f9b2ce746..c6c48fd66 100644
--- "a/mingw-w64-fltk/PKGBUILD"
+++ "b/mingw-w64-fltk/PKGBUILD"
@@ -23,11 +23,13 @@ options=('strip' 'staticlibs' 'buildflags')
 source=("${_realname}-${pkgver}.tar.gz::https://github.com/${_realname}/${_realname}/archive/release-${pkgver}.tar.gz"
         "0001-cmake-fix-lib-mess.patch"
         "0002-cmake-disable-dlfcn-search.patch"
-        "0003-cmake-fix-install.patch")
+        "0003-cmake-fix-install.patch"
+        "0004-correct-opengl-libraries-win32.patch")
 sha256sums=('5c534287b0e03b9520ff866704a5649268986b371bdf8f6ac003fa240e761901'
             'b8061f26d989f76441d68a07ba4089b50eceb90fb3234106a88511a12d9ddb50'
             '05daefb5031defa3c9fbba48c46d1322f226bd4a3e89c31a1230817ba6d6d6ed'
-            '52c1b13eaceec9fbefdd13e19cc71120033aa8168ec6e4149715eb63e74fe224')
+            '52c1b13eaceec9fbefdd13e19cc71120033aa8168ec6e4149715eb63e74fe224'
+            'b8796a094287a7cbfa88998ed3cac499420be4b20138525f631c863d9da57ce9')
 
 prepare() {
   cd "${srcdir}/${_realname}-release-${pkgver}"
@@ -39,6 +41,7 @@ prepare() {
 
   patch -p1 -i "${srcdir}/0002-cmake-disable-dlfcn-search.patch"
   patch -p1 -i "${srcdir}/0003-cmake-fix-install.patch"
+  patch -p1 -i "${srcdir}/0004-correct-opengl-libraries-win32.patch"
 }
 
 build() {
```

Re-built with `MINGW_ARCH=mingw64 makepkg-mingw -sLf` and installed with `pacman -U mingw-w64-x86_64-fltk-1.3.5-2-any.pkg.tar.zst`.
After that, I see:
```
$ fltk-config --use-gl --ldflags
-L/mingw64/lib -lfltk_gl -lglu32 -lopengl32 -lfltk -lpthread -lfontconfig
```

So afaict, your patch works perfectly.

**Comment by revelator** [comment]
your welcome :)

**Comment by mmuetzel** [comment]
Afaict, this is fixed upstream in [fltk 1.3.6](https://github.com/fltk/fltk/releases/tag/release-1.3.6).

**Comment by mmuetzel** [comment]
Thank you.
This seems to be working for me now.
